### DISCUSSION TITLE
New level system

### DISCUSSION THREAD

**Discussion by floralpetals** [eetnaviation/infratextadventurereboot#4] [Ideas]
Looking at INFRA and how there are many puzzels to be solved during a let's play, I would suggest moving puzzles to python objects instead of JSON files to allow puzzles and more advanced stuff.

**Comment by eetnaviation** [comment]
Good idea. But I really don't know how to do that. I would like more puzzles tho. It's just that implementing all the puzzles in json is possible but then for example the puzzle at hammer valley dam changes how you can play and what you can do in the bunker for example.

**Comment by floralpetals** [comment]
Yes, there would probably be a need to implement a save-system for that as well. In source INFRA uses global variables as far as I know.

**Comment by floralpetals** [comment]
I worked on a new levelling system a bit today, I can show a preview/accept feedback tomorrow.

**Reply by eetnaviation** [reply]
This would also probably mean a full redesign of the menu and save data could lrobably be stored in a csv file. Would 14th november work? Also my discord is island#2229

**Reply by floralpetals** [reply]
Yes a full redesign would be needed. I tried sending you a discord friend request but you do not accept them. Also what do you mean by "would 14th november work?"

**Reply by floralpetals** [reply]
Finally got an example,
```py
Level("office")\
  .addPosition(Position("0")\
    .addText("This will always show up.")
    .addCommand(Command(label="Set idk to true.", shortId="idk", onUse=example))
    .addCommand(Command(label="Progress", shortId="pr").moveToPosition("1"))
  )\
  .addPosition(Position("1")\
    .addTextPredicate(lambda: False, "This will never show up.")
    .addTextPredicate(lambda: idk, "This will show up once idk is True.")
    .addCommand(Command(label="Set idk to true", shortId="true", onUse=example))
    .addCommand(Command(label="Only shows if idk is true", predicate=lambda: idk).moveToPosition("0"))
  )
  ```

**Reply by floralpetals** [reply]
https://github.com/eetnaviation/infratextadventurereboot/pull/7
